capsules/extra/src/ieee802154/framer.rs

```diff
@@ -751,7 +749,7 @@ impl<'a, M: Mac<'a>, A: AES128CCM<'a>> MacDevice<'a> for Framer<'a, M, A> {
         dst_pan: PanID,
         dst_addr: MacAddress,
         src_pan: PanID,
-        mut src_addr: MacAddress,
+        mut _src_addr: MacAddress,
```

Comment:
Should this just be deleted?

Reply:
I probably should have drawn greater attention to this. I think the short answer is yes, we should remove it. To get Thread to work, I removed the call to the lookup function (for crypto nonce) and instead just used the mac addr in the packet header. Previously, the 15.4 capsule would add a neighbor (and save the short/long addr) so that upon receiving an encrypted packet with only a short src addr, the packet could be decrypted if the neighbor had been previously saved. 

A bigger potential discussion is how should 15.4 behave? I think the answer (as we've discussed) is to strictly follow the spec. I suspect we may want to remove the neighbor functionality all together and have the 15.4 capsule follow the smallest subset of the spec.

Reply:
Looking further into this today, this issue could easily be resolved by the UDP mux being given the long address when the component is created in `main.rs`. I've updated the function accordingly (now using the `src_addr` argument)